presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -754,6 +770,25 @@ private PlanBuilder appendScalarSubqueryJoin(PlanBuilder builder, SubqueryExpres
         }
     }
 
+    private ProjectNode coercionScalarSubquery(PlanNode node, SubqueryExpression scalarSubquery, RelationPlan relationPlan)
+    {
+        TranslationMap translations = new TranslationMap(relationPlan, analysis);
+        translations.setFieldMappings(relationPlan.getOutputSymbols());
+
+        Symbol output = getOnlyElement(node.getOutputSymbols());
+        translations.put(scalarSubquery, output);
+
+        PlanBuilder subPlan = new PlanBuilder(translations, relationPlan.getRoot(), relationPlan.getSampleWeight());
+
+        ImmutableMap.Builder<Symbol, Expression> projections = ImmutableMap.builder();
```

Comment:
No need to build this map. Just pass the result of coerce(...) to the Project node.